File: chatuikit/src/main/java/com/cometchat/chatuikit/shared/resources/localise/Language.java
package com.cometchat.chatuikit.shared.resources.localise;

import androidx.annotation.StringDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

/**
 * The Language class provides constants and annotations for supported language
 * codes.
 *
 * <p>
 * It defines a set of language codes as String constants and provides a
 * StringDef annotation for validation.
 */
public class Language {
    private static final String TAG = Language.class.getSimpleName();


    /**
     * Annotation for defining valid language codes.
     */
    @Retention(RetentionPolicy.SOURCE)
    @StringDef({Code.de, Code.ar, Code.zh_tw, Code.en, Code.fr, Code.es, Code.hi, Code.hu, Code.lt, Code.ms, Code.ru, Code.zh, Code.sv, Code.pt, Code.it, Code.ja, Code.ko, Code.nl, Code.tr})
    public @interface Code {
        String de = "de";
        String ar = "ar";
        String zh_tw = "zh-TW";
        String en = "en";
        String es = "es";
        String ms = "ms";
        String sv = "sv";
        String hu = "hu";
        String lt = "lt";
        String ru = "ru";
        String fr = "fr";
        String pt = "pt";
        String hi = "hi";
        String zh = "zh";
        String it = "it";
        String ja = "ja";
        String ko = "ko";
        String nl = "nl";
        String tr = "tr";
    }
}

File: chatuikit-core/src/main/java/com/cometchat/uikit/core/resources/localise/Language.kt
package com.cometchat.uikit.core.resources.localise

import androidx.annotation.StringDef

/**
 * Language codes supported by CometChat UIKit.
 * Use these constants with [CometChatLocalize.setLocale] to set the app language.
 */
object Language {
    const val ENGLISH = "en"
    const val SPANISH = "es"
    const val FRENCH = "fr"
    const val GERMAN = "de"
    const val PORTUGUESE = "pt"
    const val ITALIAN = "it"
    const val RUSSIAN = "ru"
    const val CHINESE = "zh"
    const val JAPANESE = "ja"
    const val KOREAN = "ko"
    const val ARABIC = "ar"
    const val HINDI = "hi"
    const val TURKISH = "tr"
    const val DUTCH = "nl"
    const val POLISH = "pl"
    const val SWEDISH = "sv"
    const val HUNGARIAN = "hu"
    const val MALAY = "ms"
    const val LITHUANIAN = "lt"

    /**
     * Annotation for language code validation.
     */
    @StringDef(
        ENGLISH, SPANISH, FRENCH, GERMAN, PORTUGUESE, ITALIAN,
        RUSSIAN, CHINESE, JAPANESE, KOREAN, ARABIC, HINDI,
        TURKISH, DUTCH, POLISH, SWEDISH, HUNGARIAN, MALAY, LITHUANIAN
    )
    @Retention(AnnotationRetention.SOURCE)
    annotation class Code
}
